### ISSUE TITLE
[Question] In my spring boot project config isFull is true can not register meta data

### ISSUE THREAD

**Issue by MaiSR9527** [apache/shenyu#4512]
### Question

My config application yml. I use http register type. I do not want to add `@ShenyuSpringMvcClient` on each controller method, so I try to use isFull. But meta data is not registered.
```yaml
shenyu:
  register:
    registerType: http
    serverLists: http://localhost:9095 
    props:
      username: admin
      password: 123456
  client:
      http:
        props:
          contextPath: /user
          appName: user
          isFull: true
```

log `registerMetaData` is false
```
 metadata client register success: {"appName":"user","contextPath":"/user","path":"/user/","rpcType":"http","ruleName":"/user","enabled":true,"pluginNames":[],"registerMetaData":false,"timeMillis":1680074943806,"addPrefixed":false} 
```

`org.apache.shenyu.client.springmvc.init.SpringMvcClientEventListener#getBeans`  don't set registerMetaData, default is false
```java
    @Override
    protected Map<String, Object> getBeans(final ApplicationContext context) {
        // Filter out
        if (Boolean.TRUE.equals(isFull)) {
            getPublisher().publishEvent(MetaDataRegisterDTO.builder()
                    .contextPath(getContextPath())
                    .addPrefixed(addPrefixed)
                    .appName(getAppName())
                    .path(PathUtils.decoratorPathWithSlash(getContextPath()))
                    .rpcType(RpcTypeEnum.HTTP.getName())
                    .enabled(true)
                    .ruleName(getContextPath())
                    .build());
            return null;
        }
        return context.getBeansWithAnnotation(Controller.class);
    }
```
In `org.apache.shenyu.admin.service.register.ShenyuClientRegisterDivideServiceImpl#registerMetadata`
```java
    protected void registerMetadata(final MetaDataRegisterDTO dto) {
        if (dto.isRegisterMetaData()) {
            MetaDataService metaDataService = getMetaDataService();
            MetaDataDO exist = metaDataService.findByPath(dto.getPath());
            metaDataService.saveOrUpdateMetaData(exist, dto);
        }
    }
```
I use ShenYu version 2.5.1.
So http register type do not support isFull ？？？？？？？？？？？？？

**Comment by yu199195** [comment]
why need register metadata？

**Comment by MaiSR9527** [comment]
> why need register metadata？

I do not want to add @ShenyuSpringMvcClient on each controller method. so I try to use isFull.

**Comment by MaiSR9527** [comment]
> > why need register metadata？
> 
> I do not want to add @ShenyuSpringMvcClient on each controller method. so I try to use isFull.

My project has not previously used shenyu gateway. 
So I try to use. 
But My project have so many Controller

**Comment by yu199195** [comment]
shenyu client is not must， you can config in shenyu-admin

**Comment by echojun186** [comment]
I had the same problem

**Comment by MaiSR9527** [comment]
> shenyu client is not must， you can config in shenyu-admin

I know.
But this way is too tedious and `isFull=ture` doesn't work.
Although I modify the source code like this, the request path was not saved in database
···
protected void registerMetadata(final MetaDataRegisterDTO dto) {
        // if (dto.isRegisterMetaData()) {
            MetaDataService metaDataService = getMetaDataService();
            MetaDataDO exist = metaDataService.findByPath(dto.getPath());
            metaDataService.saveOrUpdateMetaData(exist, dto);
        // }
    }
···
**I think this is a bug.** If I want this function I have to rewrite it like `SpringMvcClientEventListener`

**Comment by moremind** [comment]
maybe you can focus on `org.apache.shenyu.client.core.client.AbstractContextRefreshedEventListener`

**Comment by moremind** [comment]
> maybe you can focus on `org.apache.shenyu.client.core.client.AbstractContextRefreshedEventListener`

check the method of `getBeans`

**Comment by MaiSR9527** [comment]
> > maybe you can focus on `org.apache.shenyu.client.core.client.AbstractContextRefreshedEventListener`
> 
> check the method of `getBeans`

I konw

**Comment by moremind** [comment]
`isFull` means begin with `context-path` as match rule.

**Comment by moremind** [comment]
> `isFull` means begin with `context-path` as match rule.

for example you config `context-path: /http`, shenyu will register rule as follows:
![image](https://user-images.githubusercontent.com/20184263/228581426-b6b3946a-3799-4a02-9aa5-1e96e744bda5.png)

this isn't a bug

**Comment by MaiSR9527** [comment]
> > `isFull` means begin with `context-path` as match rule.
> 
> for example you config `context-path: /http`, shenyu will register rule as follows: ![image](https://user-images.githubusercontent.com/20184263/228581426-b6b3946a-3799-4a02-9aa5-1e96e744bda5.png)
> 
> this isn't a bug



# new test
I know I can config by shenyu admin. Why I don't use any `@ShenyuSpringMvcClient` instead of using `isFull=ture`  can create that
![image](https://user-images.githubusercontent.com/38725713/228595692-3aaca433-2a06-4f23-9f8a-4857beff8ac4.png)
![image](https://user-images.githubusercontent.com/38725713/228596747-6828ac56-c960-46b2-9fa4-1b8c19b8d697.png)


But when I access my interface it occur error. like #4514
Do I have to configure some thing in shenyu admin ？Must use `@ShenyuSpringMvcClient` ?
![image](https://user-images.githubusercontent.com/38725713/228596428-5fd2e8ba-e105-41a6-acbb-5a7a1fc2a403.png)

**Comment by moremind** [comment]
> > > `isFull` means begin with `context-path` as match rule.
> > 
> > 
> > for example you config `context-path: /http`, shenyu will register rule as follows: ![image](https://user-images.githubusercontent.com/20184263/228581426-b6b3946a-3799-4a02-9aa5-1e96e744bda5.png)
> > this isn't a bug
> 
> # new test
> I know I can config by shenyu admin. When I use `isFull=ture` instead of using `@ShenyuSpringMvcClient` can automatically create that ![image](https://user-images.githubusercontent.com/38725713/228595692-3aaca433-2a06-4f23-9f8a-4857beff8ac4.png) ![image](https://user-images.githubusercontent.com/38725713/228596747-6828ac56-c960-46b2-9fa4-1b8c19b8d697.png)
> 
> But when I access my interface it occur error. like #4514 I only use `isFull=true`. Do I have to configure some thing in shenyu admin ？Must use `@ShenyuSpringMvcClient` ? ![image](https://user-images.githubusercontent.com/38725713/228596428-5fd2e8ba-e105-41a6-acbb-5a7a1fc2a403.png)

selector must exist data, can you fix it?

**Comment by echojun186** [comment]
> 

Hi, have you solved your problem?

**Comment by MaiSR9527** [comment]
> > 
> 
> Hi, have you solved your problem?

I think need modifying source code to fix it

**Comment by yu199195** [comment]
if selector have not upstream， you can config  ip + port in selector

**Comment by MaiSR9527** [comment]
I don't want to add  `@ShenyuSpringMvcClient` on every method or config by shenyu admin, but I want to register all request path.
Maybe I will write a `Listener` to do it in my project. 😀

**Comment by echojun186** [comment]
[https://github.com/apache/shenyu/issues/4514](url) 
You maybe refer to this

**Comment by MaiSR9527** [comment]
> [https://github.com/apache/shenyu/issues/4514](url) You maybe refer to this

I know getBeans return null skip register uri. 
But I need not to use `@ShenyuSpringMvcClient` and register all `@RequestMqpping` value
I just try to use `isFull`.
Now I know how to do it

**Comment by meta-ease** [comment]
> 

I also did this, rewrote the `SpringCloudClientEventListener`.

**Comment by peijxx** [comment]
have a repair plan？ we encountering the same problem

**Comment by moremind** [comment]
> have a repair plan？ we encountering the same problem

in next week

**Comment by moremind** [comment]
Hi, it will fix in #4681
